tests/Runner/Parallel/ParallelConfigFactoryTest.php

```diff
@@ -61,4 +61,39 @@ public function testDetectConfigurationWithParams(): void
         self::assertSame(22, $config->getFilesPerProcess());
         self::assertSame(2_200, $config->getProcessTimeout());
     }
+
+    public function testDetectConfigurationWithDefaultValue(): void
+    {
+        $config = ParallelConfigFactory::detect(null, 60);
+
+        self::assertSame(10, $config->getFilesPerProcess());
+        self::assertSame(60, $config->getProcessTimeout());
+    }
+
+    /**
+     * @requires PHP 8.0
+     */
+    public function testDetectConfigurationWithNamedArgs(): void
+    {
+        // First argument omitted, second one provided via named argument
+        $config1 = \call_user_func_array([ParallelConfigFactory::class, 'detect'], ['processTimeout' => 300]);
+
+        self::assertSame(10, $config1->getFilesPerProcess());
+        self::assertSame(300, $config1->getProcessTimeout());
+
+        // Flipped order of arguments using named arguments syntax
+        $config2 = \call_user_func_array([ParallelConfigFactory::class, 'detect'], [
+            'processTimeout' => 300,
+            'filesPerProcess' => 5,
+        ]);
+
+        self::assertSame(5, $config2->getFilesPerProcess());
+        self::assertSame(300, $config2->getProcessTimeout());
+
+        // Only first argument provided, but via named argument
+        $config3 = \call_user_func_array([ParallelConfigFactory::class, 'detect'], ['filesPerProcess' => 7]);
+
+        self::assertSame(7, $config3->getFilesPerProcess());
+        self::assertSame(120, $config3->getProcessTimeout());
```

Comment:
we could use `DEFAULT_PROCESS_TIMEOUT` now, in that case.

Reply:
IMHO values in asserts should not reuse constants because then you don't catch that something has changed (assert follows const value automatically, so you always have proper value in the assert). Tests should fail when constants are changed, as this is a change in public contract. Do you have a different experience?

Reply:
if method is supposed to calculate the value, yes, it should be hardcoded/ninja value.
Yet if method is expected to return given special value (here - default value) "as is", then it's OK to rely on given constant. In such test you care if test returned given special value, not what that value actually is.

Beside of request to not keep the existence of those constants under BC promise, changing the value of the constant under BC promise is not BC break ([ref](https://symfony.com/doc/current/contributing/code/bc.html#changing-classes))